authors/apps/authentication/serializers.py

```diff
@@ -214,55 +215,129 @@ class SocialAuthSerializer(serializers.Serializer):
             raise serializers.ValidationError(e)
         try:
-            User.objects.get(email=value)
+            User.objects.get(email=email)
         except ObjectDoesNotExist:
-            raise serializers.ValidationError(
-                "no account with that email address"
-            )
+            raise serializers.ValidationError({
+                "email": "No account with that email address"
+            })
+        return data
 
 
 class PasswordResetConfirmSerializer(serializers.Serializer):
     """
     Serializer class for password reset confirm
     """
-    token = serializers.CharField(max_length=250, required=True)
-    password = serializers.CharField(max_length=250, required=True)
-    password_confirm = serializers.CharField(max_length=250, required=True)
+    token = serializers.CharField(
+        max_length=250,
+        write_only=True,
+        allow_null=True,
+        allow_blank=True,
+        required=False
+    )
+    password = serializers.CharField(
+        max_length=250,
+        write_only=True,
+        allow_null=True,
+        allow_blank=True,
+        required=False
+    )
+    password_confirm = serializers.CharField(
+        max_length=250,
+        write_only=True,
+        allow_null=True,
+        allow_blank=True,
+        required=False
+    )
 
     def validate(self, data):
+        if not data.get("token"):
+            raise serializers.ValidationError({
+                "token": "There is no token provided"
+            })
         try:
             token_object = Token.objects.get(key=data.get("token"))
         except ObjectDoesNotExist:
             raise serializers.ValidationError({
-                "token": "invalid token"
+                "token": "Invalid token"
             })
         if self.expired_token(token_object):
             raise serializers.ValidationError({
-                "token": "invalid token"
+                "token": "Expired token"
             })
+        self.get_password_field_erros(data)
         try:
             validate_password(data.get("password"))
         except ValidationError as error:
             raise serializers.ValidationError({
                 "password": list(error)
             })
-        if data["password"] != data["password_confirm"]:
+        policy = PasswordPolicy.from_names(
+            uppercase=2,
+            numbers=2,
+            special=2
+        )
+        if policy.test(data.get("password")):
+            stats = PasswordStats(data.get("password"))
+            if stats.letters_uppercase < 2:
+                raise serializers.ValidationError({
+                    "password": "Your password should have a minimum of 2 uppercase letters"
+                })
+            elif stats.numbers < 2:
+                raise serializers.ValidationError({
+                    "password": "Your password should have a minimum of 2  numbers"
+                })
+            elif stats.special_characters < 2:
```

Comment:
Same case here. Work closely with @Philipotieno to be uniform in this validations.

Reply:
He specified the special characters to be a minimum of two

Reply:
This is the policy we have settle for:
```
3 lowercase
2 uppercase
1 special character
2 digits
```
I have updated the necessary changes